tests/core/full_node/test_mempool.py

```diff
@@ -793,3 +794,211 @@ async def test_agg_sig_condition(self, two_nodes):
         #
         # sb = full_node_1.full_node.mempool_manager.get_spendbundle(spend_bundle.name())
         # assert sb is spend_bundle
+
+    @pytest.mark.asyncio
+    async def test_correct_my_parent(self, two_nodes):
+        reward_ph = WALLET_A.get_new_puzzlehash()
+        full_node_1, full_node_2, server_1, server_2 = two_nodes
+        blocks = await full_node_1.get_all_full_blocks()
+        start_height = blocks[-1].height
+        blocks = bt.get_consecutive_blocks(
+            3,
+            block_list_input=blocks,
+            guarantee_transaction_block=True,
+            farmer_reward_puzzle_hash=reward_ph,
+            pool_reward_puzzle_hash=reward_ph,
+        )
+        full_node_1, full_node_2, server_1, server_2 = two_nodes
+        peer = await connect_and_get_peer(server_1, server_2)
+
+        for block in blocks:
+            await full_node_1.full_node.respond_block(full_node_protocol.RespondBlock(block))
+
+        await time_out_assert(60, node_height_at_least, True, full_node_1, start_height + 3)
+
+        coin = list(blocks[-1].get_included_reward_coins())[0]
+        cvp = ConditionVarPair(ConditionOpcode.ASSERT_MY_PARENT_ID, [coin.parent_coin_info])
+        dic = {cvp.opcode: [cvp]}
+
+        spend_bundle1 = generate_test_spend_bundle(coin, dic)
+
+        assert spend_bundle1 is not None
+        tx1: full_node_protocol.RespondTransaction = full_node_protocol.RespondTransaction(spend_bundle1)
+        await full_node_1.respond_transaction(tx1, peer)
+
+        sb1 = full_node_1.full_node.mempool_manager.get_spendbundle(spend_bundle1.name())
+
+        assert sb1 is spend_bundle1
```

Comment:
I think it would be okay to assert sb1 == spend_bundle1

But no problem as is.